modules/reindex/src/test/java/org/elasticsearch/index/reindex/RethrottleTests.java

```diff
@@ -174,6 +171,39 @@ private void testCase(AbstractBulkByScrollRequestBuilder<?, ?> request, String a
                 response.getBatches(), greaterThanOrEqualTo(numSlices));
     }
 
+    private ListTasksResponse rethrottleTask(TaskId taskToRethrottle, float newRequestsPerSecond) {
+        // the task isn't ready to be rethrottled until it has figured out how many slices it will use. if we rethrottle when the task is
+        // in this state, the request will fail. so we try a few times
+        long start = System.nanoTime();
+        List<Throwable> failures = new ArrayList<>();
+        do {
```

Comment:
I've done:
```
assertBusy(() -> {
  try {
    stuff
  } catch (ElasticsearchException e) {
    throw new AssertionFailure("rethrowing to cause retry");
  }
});
```

It isn't always clearer, especially if you want to return something from the results of `assertBusy`, but I think if you don't return something then it probably is clearer. Probably.